hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/AccountBalanceRepository.java

```diff
@@ -42,6 +42,37 @@ insert into account_balance (account_id, balance, consensus_timestamp)
     @Transactional
     int balanceSnapshot(long consensusTimestamp);
 
+    @Override
+    @Modifying
+    @Query(
+            nativeQuery = true,
+            value =
+                    """
+        insert into account_balance (account_id, balance, consensus_timestamp)
+        select id, balance, :consensusTimestamp
+        from entity
+        where
+          id = 2 or
+          (balance is not null and
+           deleted is not true and
+           balance_timestamp > :maxConsensusTimestamp and
+           balance_timestamp < :upperRangeTimestamp)
```

Comment:
```suggestion
           balance_timestamp > :maxConsensusTimestamp)
```

Check the upper bound is unnecessary. There is an inherent on the caller to guarantee that when the method is called, the entity table state should be at the time of `consensusTimestamp`.

Since the consensusTimestamp is the new account balance snapshot's timestamp, and it's inside the current partition, it should never happen that any entity's balanceTimestamp >= upperRangeTimestamp.

Instead, we can add a javadoc  to the interface method.

Note, in the historical balance service, we set transaction isolation level to repeatable reads to guarantee it.

Reply:
Thanks I have made this change and removed the `upperRangeTimestamp`.